tensorflow/c/kernels/summary_op.cc

```diff
@@ -0,0 +1,172 @@
+
+/* Copyright 2020 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+#include <sstream>
+
+#include "tensorflow/c/kernels.h"
+#include "tensorflow/c/tf_tensor.h"
+#include "tensorflow/core/framework/selective_registration.h"
+#include "tensorflow/core/framework/summary.pb.h"
+#include "tensorflow/core/framework/types.h"
+
+// Struct that stores the status and TF_Tensor inputs to the opkernel. 
+// Used to delete tensor and status in its destructor upon kernel return. 
+typedef struct Params{ 
```

Comment:
Since this translation unit can be pure C++, let's use a normal C++ class instead; the typedef struct pattern is more of a C thing.

Also, let's define the class in an anonymous namespace as well.